assets/src/design-system/components/input/index.js

```diff
@@ -99,6 +123,11 @@ export const Input = ({
         hasError={hasError}
         {...props}
       />
+      {Icon && (
+        <IconContainer data-testid="input-icon">
```

Comment:
I don't think this test-id belongs here. If the icon has semantic meaning, it should have a label inspectable by both screen readers and testing-library. If not, it should not be tested by testing-library.

I do think most suffix icons **should** have a label, but that's up to the outer component to set, not the input field itself (as it doesn't know what's in the icon nor what it means).